test/deals.js

```diff
@@ -140,24 +260,82 @@ describe('Deals', function() {
           ],
         })
         .then(data => {
+          dealId = data.dealId
           expect(data).to.be.a('object')
         })
     })
   })
 
-  // describe('Associate', function () {
-  //   it('Returns a 204 response if successful.', function () {
-  //     return hubspot.deals.associate(1126609, 'CONTACT', 394455).then(data => {
-  //       expect(data).to.be.a('object')
-  //     })
-  //   })
-  // })
-
-  // describe('Remove Association', function () {
-  //   it('Returns a 200 response if successful.', function () {
-  //     return hubspot.deals.removeAssociation(1126609, 'CONTACT', 394455).then(data => {
-  //       expect(res.statusCode).to.equal(204)
-  //     })
-  //   })
-  // })
+  describe('Associate', function() {
+    let dealId = 123
+    let companyId = 234
+    const associationType = 'COMPANY'
+    const associateDealEndpoint = {
+      path: `/deals/v1/deal/${dealId}/associations/${associationType}`,
+      query: { id: companyId },
+      statusCode: 204,
+    }
+    fakeHubspotApi.setupServer({ putEndpoints: [associateDealEndpoint] })
+
+    before(function() {
+      if (process.env.NOCK_OFF) {
+        return Promise.all([
+          createTestDeal().then(data => (dealId = data.dealId)),
+          createTestCompany().then(data => (companyId = data.companyId)),
+        ])
+      }
+    })
+    after(function() {
+      if (process.env.NOCK_OFF) {
+        return Promise.all([
+          deleteTestDeal(dealId),
+          deleteTestCompany(companyId),
+        ])
+      }
+    })
+
+    it('Returns a 204 response if successful.', function() {
```

Comment:
Since this is the successful path, and there's not a test for the unsuccessful path, what do you think of shortening this description to "Returns a 204"? I suspect you included "if successful" to highlight that a successful response does indeed return a 204 (which is a bit odd in this case) but it makes me look for a test that exercises the unsuccessful path.